streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -510,64 +503,49 @@ StandbyTask createTask(final Consumer<byte[], byte[]> consumer,
-            punctuateTimeSensor.add(metrics.metricName("punctuate-latency-max", this.groupName, "The maximum punctuate time in ms", this.tags), new Max());
-            punctuateTimeSensor.add(createMeter(metrics, new Count(), "punctuate", "punctuate calls"));
-
-            taskCreatedSensor = metrics.sensor(prefix + ".task-created", Sensor.RecordingLevel.INFO);
-            taskCreatedSensor.add(createMeter(metrics, new Count(), "task-created", "newly created tasks"));
-
-            tasksClosedSensor = metrics.sensor(prefix + ".task-closed", Sensor.RecordingLevel.INFO);
-            tasksClosedSensor.add(createMeter(metrics, new Count(), "task-closed", "closed tasks"));
-
-            skippedRecordsSensor = metrics.sensor(prefix + ".skipped-records");
-            skippedRecordsSensor.add(createMeter(metrics, new Sum(), "skipped-records", "skipped records"));
 
-        }
-
-        private Meter createMeter(final Metrics metrics,
-                                  final SampledStat stat,
-                                  final String baseName,
-                                  final String descriptiveName) {
-            final MetricName rateMetricName = metrics.metricName(
-                baseName + "-rate",
-                groupName,
-                String.format("The average per-second number of %s", descriptiveName),
-                tags
-            );
-            final MetricName totalMetricName = metrics.metricName(
-                baseName + "-total",
-                groupName,
-                String.format("The total number of %s", descriptiveName),
-                tags
-            );
-            return new Meter(stat, rateMetricName, totalMetricName);
+        private final Sensor commitTimeSensor;
+        private final Sensor pollTimeSensor;
+        private final Sensor processTimeSensor;
+        private final Sensor punctuateTimeSensor;
+        private final Sensor taskCreatedSensor;
+        private final Sensor tasksClosedSensor;
+
+        StreamsMetricsThreadImpl(final Metrics metrics, final String threadName) {
+            super(metrics, threadName);
+            final String groupName = "stream-metrics";
+
+            commitTimeSensor = metrics.sensor(threadLevelSensorName(threadName, "commit-latency"), Sensor.RecordingLevel.INFO);
+            commitTimeSensor.add(metrics.metricName("commit-latency-avg", groupName, "The average commit time in ms", tags()), new Avg());
+            commitTimeSensor.add(metrics.metricName("commit-latency-max", groupName, "The maximum commit time in ms", tags()), new Max());
+            commitTimeSensor.add(metrics.metricName("commit-rate", groupName, "The average per-second number of commit calls", tags()), new Rate(TimeUnit.SECONDS, new Count()));
+            commitTimeSensor.add(metrics.metricName("commit-total", groupName, "The total number of commit calls", tags()), new Count());
```

Comment:
I refactored these to flatten the metric definition, since it was super hard to figure out what metrics were actually being created.

Maybe you can forgive me for this because I actually found a bug: The description of the total metrics say that it counts the number of calls, but it previously summed the recorded values. (was via createMeter -> new Meter -> new Total)

Reply:
Good call!